### PR TITLE
feat(FEC-12037): [SmartSwitch] Mediacorp Multi CDN POC add mainfest redirect config

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Add config option `followRedirects` config option 
And add manifestUrl redirect handler which executed when followRedirects set to true;

solves: FEC-12037

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/smart-switch.js [hunk 1]
```diff
@@ -104,6 +109,12 @@ class SmartSwitch extends BasePlugin {
     return Utils.Http.execute(url, null, 'GET', null, this._responseTimeoutMs, responseTimeoutHandler);
   }
 
+  async _extractResource(resourceUrl: string) {
+    const data = await fetch(resourceUrl);
+    console.log(data);
```
Comment (SivanA-Kaltura) [inline src/smart-switch.js:114]: use logger

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/smart-switch.js [hunk 2]
```diff
@@ -84,11 +85,15 @@ class SmartSwitch extends BasePlugin {
...
   }
 
-  _doRequest(resource: string): Promise<any> {
+  async _doRequest(resource: string): Promise<any> {
     const responseTimeoutHandler = () => {
       this.logger.warn(`Timeout reached ${this.config.responseTimeoutSec} seconds, loading original source`);
     };
 
+    if (this.config.followRedirects) {
+      resource = await this._extractResource(resource);
```
Comment (yairans) [inline src/smart-switch.js:94]: it should run recursively to get the final resources in case of multi redirections

**Review by yairans — CHANGES_REQUESTED** [review]

File: src/smart-switch.js (hunk 2 above)
Comment (JonathanTGold) [reply src/smart-switch.js:94]: it done automatically by the browser

File: src/smart-switch.js (hunk 1 above)
Comment (JonathanTGold) [reply src/smart-switch.js:114]: removed

Comment (yairans) [reply src/smart-switch.js:114]: is not :)

Comment (JonathanTGold) [reply src/smart-switch.js]: fixed

File: src/smart-switch.js (hunk 2 above)
Comment (yairans) [reply src/smart-switch.js:94]: per [mdn](https://developer.mozilla.org/en-US/docs/Web/API/fetch) it's not supported in ie11. I guess there is polyfill you can use

**Review by yairans — CHANGES_REQUESTED** [review]

**Review by yairans — APPROVED** [review]

File: src/smart-switch.js (hunk 2 above)
Comment (JonathanTGold) [reply src/smart-switch.js:94]: polyfill added

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**